### DISCUSSION TITLE
Missing property for nested write with createMany and composite IDs

### DISCUSSION THREAD

**Discussion by pepoospina** [prisma/prisma#14328] [Q&A]
My schema is 
```
model Campaign {
  uri             String    @id
  roots           CampaignRoot[] 
}

model CampaignRoot {
  @@id([campaignId, root])
  campaign        Campaign  @relation(fields: [campaignId], references: [uri])
  campaignId      String
  root            String
  date            BigInt
  balances        BalanceLeaf[]   // Leafs are stored on a table to be efficiently read
}

model BalanceLeaf {
  @@id([rootId, address])
  root            CampaignRoot  @relation(fields: [rootId, address], references: [campaignId, root])
  rootId          String
  address         String
  balance         String
  proof           String[]      // Store the proofs so that they don't have to be recomputed eveytime they are needed
}
```

I want to create a new CampaignRoot with many BalanceLeaf for an existing Campaign. I tried
```ts
prisma.campaignRoot.create({
      data: {
        campaignId: uri,
        date,
        root,
        balances: {
          createMany: {
            data: leafs,
          },
        },
      },
    });
```

where the `leafs` array has the `address`, the `balance` and the `proof` properties. But, prisma complains that property `address` should not be there.

In fact, checking the types from prisma we can see that the expected type for a nested write does not have the address:

```ts
  export type BalanceLeafCreateWithoutRootInput = {
    balance: string
    proof?: BalanceLeafCreateproofInput | Enumerable<string>
  }
```

How can I create nested elements without providing the address property? It seems to me there must be an error somewhere.

Maybe this is because the `address` is part of the composite id? Maybe I'm not configuring the nested composite IDs correctly?

**Comment by pepoospina** [comment]
So, I was making two mistakes:

The `fields` in the `BalanceLeaf` `@relation` to `CampaignRoot` should not have included the `address`, but the `campaignId`. Essentially, composite IDs must be included in all "derived" models.

Second, the composite id of the `BalanceLeaf` must also include the `campaignId`, to guarantee uniqueness.

So the schema is now


```
model CampaignRoot {
  @@id([campaignId, root])
  campaign        Campaign  @relation(fields: [campaignId], references: [uri])
  campaignId      String
  root            String
  date            BigInt
  balances        BalanceLeaf[]   // Leafs are stored on a table to be efficiently read
}

model BalanceLeaf {
  @@id([campaignId, rootId, address])
  root            CampaignRoot  @relation(fields: [campaignId, rootId], references: [campaignId, root])
  campaignId      String
  rootId          String
  address         String
  balance         String
  proof           String[]      // Store the proofs so that they don't have to be recomputed eveytime they are needed
}

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
